crates/runtime/src/embeddings/execution_plan.rs

```diff
@@ -536,10 +628,34 @@ async fn get_vectors_with_chunker(
     Ok((vectors, content_offsets))
 }
 
+fn build_embedding_pool(
+    model_parallelism: Option<usize>,
+) -> Result<ThreadPool, Box<dyn std::error::Error + Send + Sync>> {
+    let sys_parallelism = thread::available_parallelism();
+
+    let parallelism = match (model_parallelism, sys_parallelism) {
+        (Some(0), Ok(p)) => p.get(),
+        (Some(0), _) => unreachable!("Must determine system parallelism"),
```

Comment:
Should we pick a default parallelism and log a warning if this happens? Looking at the documentation for `thread::available_parallelism` it seems like it could fail for a variety of reasons